sw/host/opentitantool/src/command/image.rs

```diff
@@ -162,36 +159,97 @@ impl CommandDispatch for ManifestUpdateCommand {
                 rsa_private_key = Some(private);
             }
         }
-        let mut spx_private_key: Option<SpxKey> = None;
+
+        // A u8 buffer with u32 alignment to be able to deserialize bytes for extensions.
+        // Note: This should ideally be based on the alignment requirements of the extensions but
+        // just using `u32` works since all extensions we have are u32-aligned. We can't miss If
+        // this doesn't hold since `LayoutVerified::new()` errors out.
+        #[repr(C)]
+        pub struct Buffer32 {
+            pub bytes: [u8; image::Image::MAX_SIZE],
+            _align: [u32; 0],
+        }
+        let mut buf = Buffer32 {
+            bytes: [0; image::Image::MAX_SIZE],
+            _align: [],
+        };
+
+        let mut spx_private_key: Option<SpxKeypair> = None;
         if let Some(key) = &self.spx_key {
             let key = spx::load_spx_key(key)?;
-            image.update_spx_key(&key)?;
```

Comment:
Can we keep the below logic inside of `Image::update_spx_key` along with the `Buffer32` definition? This would remove the  need to import these constants from `Image` and the `pub` modifiers can get dropped from `Image::MAX_SIZE` and `Image::size`.

Reply:
Do you mind taking a second pass at this when you work on the extensions support? We need to update `signed_region_end`, `extension_table` and `spx_key` before we can generate `spx_signature`. That's why I just added the basic building blocks to `image`. But I think there is no harm in moving these to dedicated methods